sky/serve/autoscalers.py

```diff
@@ -132,93 +139,103 @@ def collect_request_information(
                                    current_time - self.rps_window_size)
         self.request_timestamps = self.request_timestamps[index:]
 
-    def evaluate_scaling(
-        self,
-        replica_infos: List['replica_managers.ReplicaInfo'],
-    ) -> AutoscalerDecision:
+    def _get_desired_num_replicas(self) -> int:
 
-        num_replicas = len(replica_infos)
+        # If target_qps_per_replica is not set
+        # Return default target_num_replicas.
+        if self.target_qps_per_replica is None:
+            return self.target_num_replicas
 
         # Convert to requests per second.
         num_requests_per_second = len(
             self.request_timestamps) / self.rps_window_size
-        # Edge case: num_replicas is zero.
-        requests_per_replica = (num_requests_per_second / num_replicas
-                                if num_replicas else num_requests_per_second)
-
-        logger.info(f'Requests per replica: {requests_per_replica}, '
-                    f'upper_threshold: {self.upper_threshold}, '
-                    f'lower_threshold: {self.lower_threshold}')
-        logger.info(f'Number of replicas: {num_replicas}')
-
-        target_num_replicas = num_replicas
-        if num_replicas < self.min_replicas:
-            target_num_replicas = self.min_replicas
-        elif (self.upper_threshold is not None and
-              requests_per_replica > self.upper_threshold):
-            scale_target = requests_per_replica / self.upper_threshold
-            target_num_replicas = int(scale_target * num_replicas)
-        elif (self.lower_threshold is not None and
-              requests_per_replica < self.lower_threshold):
-            scale_target = requests_per_replica / self.lower_threshold
-            target_num_replicas = int(scale_target * num_replicas)
-
+        target_num_replicas = math.ceil(num_requests_per_second /
+                                        self.target_qps_per_replica)
         target_num_replicas = max(self.min_replicas,
                                   min(self.max_replicas, target_num_replicas))
+        logger.info(f'Requests per second: {num_requests_per_second}, '
+                    f'Current target number of replicas: {target_num_replicas}')
 
-        logger.info(f'Target number of replicas: {target_num_replicas}, '
-                    f'min_replicas: {self.min_replicas}, '
-                    f'max_replicas: {self.max_replicas}')
-
-        num_replicas_delta = 0
-        if num_replicas < self.min_replicas:
-            num_replicas_delta = self.min_replicas - num_replicas
-            self.upscale_counter = 0
-        elif num_replicas > self.max_replicas:
-            num_replicas_delta = self.max_replicas - num_replicas
-            self.downscale_counter = 0
-        elif target_num_replicas > num_replicas:
+        if not self.boostrap_done:
+            self.boostrap_done = True
+            return target_num_replicas
```

Comment:
Should we always scale to `min_replica` if the current alive replicas are lower than `min_replicas`, not only the first time? Vise versa for the case when the current alive replicas are higher than `max_replicas`.

Reply:
This should be enforced under `_get_desired_num_replicas`.

Reply:
I mean, we should not wait for upscale delay seconds if the current alive replicas are lower than min_replicas. IIUC the `_get_desired_num_replicas` only enforced `min_replicas <= target_num_replicas <= max_replicas`.

Reply:
But this situation will not happen? 
`target_num_replicas` is always `min_replicas <= target_num_replicas <= max_replicas`.
So if the current alive replicas are lower than `min_replicas`, it will always scale to `target_num_replicas` which is `>= min_replicas`?